coalib/coala.py

```diff
@@ -33,20 +34,25 @@ def main():
 
     console_printer = ConsolePrinter()
     if args.show_bears or args.show_all_bears:
-        log_printer = LogPrinter(console_printer)
-        sections, _ = load_configuration(arg_list=None, log_printer=log_printer)
-        if args.show_all_bears:
-            local_bears, global_bears = collect_all_bears_from_sections(
-                sections, log_printer)
-        else:
-            # We ignore missing settings as it's not important.
-            local_bears, global_bears = fill_settings(
-                sections,
-                acquire_settings=lambda *args, **kwargs: {},
-                log_printer=log_printer)
-        show_bears(local_bears, global_bears, args.show_all_bears,
-                   console_printer)
-        return 0
+        exitcode = 0
+        try:
+            log_printer = LogPrinter(console_printer)
+            sections, _ = load_configuration(arg_list=None,
+                                             log_printer=log_printer)
+            if args.show_all_bears:
+                local_bears, global_bears = collect_all_bears_from_sections(
+                    sections, log_printer)
+            else:
+                # We ignore missing settings as it's not important.
+                local_bears, global_bears = fill_settings(
+                    sections,
+                    acquire_settings=lambda *args, **kwargs: {},
+                    log_printer=log_printer)
+            show_bears(local_bears, global_bears, args.show_all_bears,
+                       console_printer)
+        except BaseException as exception:  # pylint: disable=broad-except
+            exitcode = exitcode or get_exitcode(exception, log_printer)
```

Comment:
exitcode will always be zero here, just `return get_exitcode(...)` here and simplify above and below accordingly